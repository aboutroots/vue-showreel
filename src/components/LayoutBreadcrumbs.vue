<template>
  <div class="LayoutBreadcrumbs breadcrumbs">
    <router-link to="/">
      <ios-home-icon class="icon" w="16" h="16" />
    </router-link>
    <span class="separator">/</span>
    {{ routeName }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'LayoutBreadcrumbs',
  computed: {
    routeName(): string {
      return this.$route.name ?? '';
    },
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';
.breadcrumbs {
  flex: 1;
  display: flex;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  margin-bottom: 0.5rem;
}
.icon {
  color: $light;
}
.separator {
  margin: 0 0.5rem;
}
</style>
