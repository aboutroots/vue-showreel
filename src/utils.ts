export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
