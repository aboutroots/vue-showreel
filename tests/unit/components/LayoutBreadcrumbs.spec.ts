import LayoutBreadcrumbs from '@/components/LayoutBreadcrumbs.vue';
import { shallowMount } from '../utils';

describe('LayoutBreadcrumbs tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(LayoutBreadcrumbs);

    const container = wrapper.find('.LayoutBreadcrumbs');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(LayoutBreadcrumbs);
    expect(wrapper.element).toMatchSnapshot();
  });
});
