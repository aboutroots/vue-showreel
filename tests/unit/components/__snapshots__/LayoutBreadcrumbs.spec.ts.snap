// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutBreadcrumbs tests suite matches snapshot 1`] = `
<div
  class="LayoutBreadcrumbs breadcrumbs"
>
  <router-link-stub
    ariacurrentvalue="page"
    event="click"
    tag="a"
    to="/"
  >
    <ios-home-icon-stub
      animate=""
      class="icon"
      h="16"
      rootclass=""
      title=""
      w="16"
    />
  </router-link-stub>
   
  <span
    class="separator"
  >
    /
  </span>
  
  

</div>
`;
